src/main/java/org/tugraz/sysds/runtime/data/DenseBlockLFP32.java

```diff
@@ -0,0 +1,331 @@
+/*
+ * Modifications Copyright 2018 Graz University of Technology
+ * 
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * 
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+package org.tugraz.sysds.runtime.data;
+
+import org.tugraz.sysds.common.Warnings;
+import org.tugraz.sysds.runtime.util.DataConverter;
+import org.tugraz.sysds.runtime.util.UtilFunctions;
+
+import java.util.Arrays;
+
+public class DenseBlockLFP32 extends DenseBlockLDRB
+{
+	private static final long serialVersionUID = 2604223782138590322L;
+
+	private float[][] _blocks;
+
+	public DenseBlockLFP32(int[] dims) {
+		super(dims);
+		reset(_rlen, _odims, 0);
+	}
+	
+	@Override
+	public boolean isNumeric() {
+		return true;
+	}
+
+	@Override
+	public boolean isContiguous() {
+		return _blocks.length == 1;
+	}
+
+	private boolean isReusable(int rlen, int[] odims) {
```

Comment:
While thinking about this notion of reusable, I remembered why before we only checked for exact matches: if we reuse too large blocks, the memory requirements of operations change and we might run out of memory although we explicitly scheduled an operation in memory. Accordingly, let's start with a more conservative approach to reusing. This might also make it much simpler such that we can pull it into the DenseBlock implementation.